src/indexer/utils/substrate/index.ts

```diff
@@ -0,0 +1,334 @@
+  const timestamp = Number((await at.query.timestamp.now()).toString())
+  const allRecords = (await at.query.system.events()) as unknown as Array<SubstrateEvent>
+
+  // we get the proposal execution events
+  const proposalExecutionEvents = getSubstrateEvents(SygmaPalleteEvents.ProposalExecution, allRecords) as Array<ProposalExecutionEvent>
+
+  // we get the deposit events - ts-ignore because of allRecords
+  const depositEvents = getSubstrateEvents(SygmaPalleteEvents.Deposit, allRecords) as Array<DepositEvent>
+
+  const failedHandlerExecutionEvents = getSubstrateEvents(SygmaPalleteEvents.FailedHandlerExecution, allRecords) as Array<FailedHandlerExecutionEvent>
+
+  // we get the index of the section in the extrinsic
+  const sectionIndex = signedBlock.block.extrinsics.findIndex(ex => ex.method.section === "sygmaBridge")
+
+  // this is our identifier for the tx
+  const txIdentifier = `${block}-${sectionIndex}` //this is like the txHash but for the substrate
+
+  proposalExecutionEvents.forEach(async (proposalExecutionEvent: ProposalExecutionEvent) => {
+    const { data } = (proposalExecutionEvent.event as any).toHuman()
+
+    const { originDomainId, depositNonce } = data
+
+    try {
+      await saveProposalExecutionToDb(
+        domain,
+        block.toString(),
+        {
+          originDomainId,
+          depositNonce: depositNonce,
+          txIdentifier,
+          blockNumber: `${block}`,
+          timestamp,
+        },
+        executionRepository,
+        transferRepository,
+        domainRepository,
+      )
+    } catch (e) {
+      logger.error(`Error saving proposal execution to db: ${e}`)
+    }
+  })
+
+  depositEvents.forEach(async (depositEvent: DepositEvent) => {
+    const { data } = (depositEvent.event as any).toHuman()
+
+    const { destDomainId, resourceId, depositNonce, sender, transferType, depositData, handlerResponse } = data
+
+    await saveDepositToDb(
+      domain,
+      block.toString(),
+      {
+        destDomainId,
+        resourceId,
+        depositNonce: depositNonce,
+        sender,
+        transferType,
+        depositData,
+        handlerResponse,
+        txIdentifier,
+        blockNumber: `${block}`,
+        timestamp,
+      },
+      transferRepository,
+      depositRepository,
+      domainRepository,
+    )
+  })
+
+  failedHandlerExecutionEvents.forEach((failedHandlerExecutionEvent: FailedHandlerExecutionEvent) => {
+    const { data } = (failedHandlerExecutionEvent.event as any).toHuman() as FailedHandlerExecutionEvent["event"]
+
+    const { originDomainId, depositNonce, error } = data
+
+    saveFailedHandlerExecutionToDb(
+      domain,
+      block.toString(),
+      {
+        originDomainId,
+        depositNonce: depositNonce,
+        error,
+        txIdentifier,
+        blockNumber: `${block}`,
+        timestamp,
+      },
+      executionRepository,
+      transferRepository,
+      domainRepository,
+    )
+  })
+}
```

Comment:
You should update domain block here, and not in `saveFailedHandlerExecutionToDb`, `saveDepositToDb`, `saveProposalExecutionToDb`, so that we update the domain block number even if the block has no events
Also it would be nice if you add some log here, something like ` logger.info(`save block on ${domain.name}: ${block.toString()}, domainID: ${}`)
```

Reply:
done